bin/build.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+config_dir="${PWD}/config"
+timestamp=$(date -u +"%Y%m%d%H%M%S")
+
+while IFS=$',' read -r board shield; do
+    extra_cmake_args=${shield:+-DSHIELD="$shield"}
+    artifact_name=${shield:+${shield// /-}-}${board}-zmk
+    filename="firmware/${timestamp}-${artifact_name}"
+    build_dir="build/${artifact_name}"
+
+    echo ""
+    echo "-----------------"
+    echo "BUILDING FIRMWARE"
+    echo "-----------------"
+    echo "Zephyr: ${ZEPHYR_VERSION}"
+    echo "Board: ${board}"
+    echo "Shield: ${shield}"
+    echo ""
+
+    if [[ ! -d "$build_dir" ]]; then
+        west build -s zmk/app -b "$board" -d ${build_dir} -- \
+            -DZMK_CONFIG="$config_dir" "${extra_cmake_args}"
+    else
+        west build -d ${build_dir} -- -DZMK_CONFIG="$config_dir"
+    fi
+
+    if [[ ! -z $OUTPUT_ZMK_CONFIG ]]; then
+        grep -v -e "^#" -e "^$" ${build_dir}/zephyr/.config \
+            | sort > "${filename}.config"
+    fi
+
+    cp ${build_dir}/zephyr/zmk.uf2 "${filename}.uf2"
```

Comment:
Would be nice to also handle `hex` files for anyone trying to use this for things like Planck, Preonic, etc. stm32 builds.

Reply:
The Github Actions workflow checks for `bin` as a fallback extension, I've updated the `build.sh` to check for `uf2`, `hex`, `bin` (in that order).